src/rgw/rgw_object_lock.cc

```diff
@@ -0,0 +1,96 @@
+#include "rgw_object_lock.h"
+
+void DefaultRetention::decode_xml(XMLObj *obj) {
+  RGWXMLDecoder::decode_xml("Mode", mode, obj, true);
+  if (mode.compare("GOVERNANCE") != 0 && mode.compare("COMPLIANCE") != 0) {
+    throw RGWXMLDecoder::err("bad Mode in lock rule");
+  }
+  bool days_exist = RGWXMLDecoder::decode_xml("Days", days, obj);
+  bool years_exist = RGWXMLDecoder::decode_xml("Years", years, obj);
+  if ((days_exist && years_exist) || (!days_exist && !years_exist)) {
+    throw RGWXMLDecoder::err("either Days or Years must be specified, but not both");
```

Comment:
Is this correct? According to https://docs.aws.amazon.com/AmazonS3/latest/API/API_DefaultRetention.html `Days` and `Years` is optional.

As far as I understood the documentation `GOVERNANCE` means [legal hold](https://docs.aws.amazon.com/AmazonS3/latest/dev/object-lock.html). I might be wrong because the docs are really bad related this issue. But if my assumption is right, the retention period has no ending which makes `Days` and `Years` useless in that case.